AEPTarget/Sources/Target.swift

```diff
@@ -428,6 +436,81 @@ public class Target: NSObject, Extension {
+        let identitySharedState = getSharedState(extensionName: TargetConstants.Identity.EXTENSION_NAME, event: event)?.value
+
+        let lifecycleContextData = getLifecycleDataForTarget(lifecycleData: lifecycleSharedState)
+
+        // set notification id, timestamp and mbox
+        let id = notificationData[TargetConstants.EventDataKeys.NOTIFICATION_ID] as? String ?? UUID().uuidString
+        let timestamp = notificationData[TargetConstants.EventDataKeys.NOTIFICATION_TIMESTAMP] as? Int64 ?? Int64(event.timestamp.timeIntervalSince1970 * 1000.0)
+
+        guard let mboxName = notificationData[TargetConstants.EventDataKeys.MBOX_NAME] as? String,
+            !mboxName.isEmpty
+        else {
+            Log.error(label: Target.LOG_TAG, "Failed to send raw Target notification, the provided mbox name is nil or empty.")
+            return
+        }
+        let mbox = Mbox(name: mboxName)
+
+        // set notification token
+        guard let tokens = notificationData[TargetConstants.EventDataKeys.NOTIFICATION_TOKENS] as? [String],
+              !tokens.isEmpty
+        else {
+            Log.warning(label: Target.LOG_TAG, "Failed to send raw Target notification, the provided tokens array is nil or empty.")
+            return
+        }
+
+        // Set notification parameters
+        let parameters = notificationData[TargetConstants.EventDataKeys.MBOX_PARAMETERS] as? [String: String]
+        let mboxParameters = Dictionary.merge(lifecycleContextData, to: parameters)
+
+        let profileParameters = notificationData[TargetConstants.EventDataKeys.PROFILE_PARAMETERS] as? [String: String]
+
+        let order = TargetOrder.from(dictionary: notificationData[TargetConstants.EventDataKeys.ORDER_PARAMETERS] as? [String: Any])?.toInternalOrder()
+        let product = TargetProduct.from(dictionary: notificationData[TargetConstants.EventDataKeys.PRODUCT_PARAMETERS] as? [String: Any])?.toInternalProduct()
+
+        let notification = Notification(
+            id: id,
+            timestamp: timestamp,
+            type: TargetConstants.TargetJson.MetricType.CLICK,
+            mbox: mbox,
+            tokens: tokens,
+            parameters: Dictionary.merge(targetParameters?.parameters, to: mboxParameters),
+            profileParameters: Dictionary.merge(targetParameters?.profileParameters, to: profileParameters),
+            order: targetParameters?.order != nil ? targetParameters?.order?.toInternalOrder() : order,
+            product: targetParameters?.product != nil ? targetParameters?.product?.toInternalProduct() : product
+        )
+        targetState.addNotification(notification)
+
+        let error = sendTargetRequest(event, targetParameters: nil, lifecycleData: lifecycleSharedState, identityData: identitySharedState) { connection in
+            self.processNotificationResponse(event: event, connection: connection)
+        }
+
+        if let err = error {
```

Comment:
nit: `if error != nil {`

Reply:
It needs to be unwrapped as well. Fixed it a little differently, however couldn't use trailing closure syntax as it introduces ambiguity.
<img width="1271" alt="Screen Shot 2022-09-26 at 7 07 20 PM" src="https://user-images.githubusercontent.com/13000319/192414754-46aa362e-b4e4-467e-a58a-496e2fc7d7ec.png">

Reply:
hmm, what do you think about this new syntax?

Reply:
Nope!